src/core/polyhedral/codegen_llvm.cc

```diff
@@ -294,6 +303,88 @@ class CodeGen_TC : public Halide::Internal::CodeGen_X86 {
 
     value = sym_get(name);
   }
+
+ public:
+  void optimize_module() {
+    Halide::Internal::debug(3) << "Optimizing module\n";
```

Comment:
Why use Halide's internal debug printer when we have our glog + flags?

Reply:
This function was taken verbatim from Halide [with the addition of one line]. Unfortunately, right now we need to do this but hopefully at one point Halide proper will support Tapir-like optimization [or permit me to just add the one line].

If you think it better, I can change to use our logging.

Reply:
I think it's better to use our own logging system. One of many reasons is also to control how we log things to terminal for python/C++ backend which right now needs to be solidified.